client/concordclient/include/client/concordclient/concord_client.hpp

```diff
@@ -48,17 +49,30 @@ struct BftTopology {
   uint16_t c_val;
   std::vector<ReplicaInfo> replicas;
   bft::client::RetryTimeoutConfig client_retry_config;
+  std::uint16_t client_sends_request_to_all_replicas_first_thresh;
+  std::uint16_t client_sends_request_to_all_replicas_period_thresh;
+  std::uint32_t client_proxies_per_replica;
+  std::string signing_key_path;
+  std::uint32_t external_requests_queue_size;
+  std::uint32_t trace_sampling_rate;
+  bool encrypted_config_enabled;
+  bool transaction_signing_enabled;
+  bool client_batching_enabled;
+  size_t client_batching_max_messages_nbr;
+  std::uint64_t client_batching_flush_timeout_ms;
 };
 
 struct BftClientInfo {
   // ID needs to match the values set in Concord's configuration
   bft::client::ClientId id;
+  uint16_t port;
+  std::string host;
 };
 
 struct TransportConfig {
   enum CommunicationType { Invalid, TlsTcp, PlainUdp };
   CommunicationType comm_type;
-
+  bool enable_mock_comm;
```

Comment:
Please add a comment that this is for testing purposes only. Ideally, we construct our classes so that production code doesn't have to deal with test code.

Reply:
done.